applications/OptimizationApplication/tests/test_mass_response_function.py

```diff
@@ -0,0 +1,254 @@
+
+import KratosMultiphysics as Kratos
+import KratosMultiphysics.OptimizationApplication as KratosOA
+
+import KratosMultiphysics.KratosUnittest as kratos_unittest
+from KratosMultiphysics.OptimizationApplication.responses.mass_response_function import MassResponseFunction
+
+class TestMassResponseFunctionBase(kratos_unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.model = Kratos.Model()
+        cls.model_part = cls.model.CreateModelPart("test")
+
+        cls.response_function = MassResponseFunction(cls.model, cls.GetParameters(), None)
+        cls.CreateElements()
+
+        cls.response_function.ExecuteInitialize()
+        cls.response_function.Check()
+        cls.response_function.ExecuteInitializeSolutionStep()
+        cls.ref_value = cls.response_function.CalculateValue()
+
+    def _CalculateSensitivity(self, sensitivity_variable):
+        self.response_function.CalculateSensitivity(sensitivity_variable, self.model_part)
+
+    def _CheckSensitivity(self, response_function, entities, sensitivity_method, update_method, delta, precision):
+        ref_value = response_function.CalculateValue()
+        for entity in entities:
+            v = sensitivity_method(entity)
+            update_method(entity, delta)
+            value = response_function.CalculateValue()
+            sensitivity = (value - ref_value)/delta
+            update_method(entity, -delta)
+            self.assertAlmostEqual(v, sensitivity, precision)
+
+    def _UpdateProperties(self, variable, entity, delta):
```

Comment:
```suggestion
    def _IncreasePropertiesBy(self, variable, entity, delta):
```
For me "update" suggests that the value gets override.

Reply:
The value is actually getting overriden by the update. `IncreasePropertiesBy` will be kind of misleading, because the properties can increase or decrease depending on the optimization outcome.